To start with, I'd like to understand what are the data I was provided with. 
Printing head of tables allow me to get to know my dataset.

In [17]:
import pandas as pd
import numpy as np 

In [18]:
artworks_python=pd.read_csv(r'C:\Users\Administrateur\Documents\artworks.csv')
artworks_python.head()

Artwork ID                                              Title Artist ID  \
0           2  Ferdinandsbrücke Project, Vienna, Austria, Ele...      6210   
1           3  City of Music, National Superior Conservatory ...      7470   
2           4  Villa near Vienna Project, Outside Vienna, Aus...      7605   
3           5  The Manhattan Transcripts Project, New York, N...      7056   
4           6  Villa, project, outside Vienna, Austria, Exter...      7605   

                       Name  Date  \
0               Otto Wagner  1896   
1  Christian de Portzamparc  1987   
2                Emil Hoppe  1903   
3           Bernard Tschumi  1980   
4                Emil Hoppe  1903   

                                              Medium  \
0      Ink and cut-and-pasted painted pages on paper   
1                  Paint and colored pencil on print   
2  Graphite, pen, color pencil, ink, and gouache ...   
3  Photographic reproduction with colored synthet...   
4  Graphite, color pencil, ink, and gouache on tr...   

                           Dimensions Acquisition Date  \
0  19 1/8 x 66 1/2" (48.6 x 168.9 cm)       1996-04-09   
1       16 x 11 3/4" (40.6 x 29.8 cm)       1995-01-17   
2   13 1/2 x 12 1/2" (34.3 x 31.8 cm)       1997-01-15   
3           20 x 20" (50.8 x 50.8 cm)       1995-01-17   
4    15 1/8 x 7 1/2" (38.4 x 19.1 cm)       1997-01-15   

                                              Credit Catalogue  ...  \
0  Fractional and promised gift of Jo Carole and ...         Y  ...   
1  Gift of the architect in honor of Lily Auchinc...         Y  ...   
2             Gift of Jo Carole and Ronald S. Lauder         Y  ...   
3  Purchase and partial gift of the architect in ...         Y  ...   
4             Gift of Jo Carole and Ronald S. Lauder         Y  ...   

  Classification Object Number Diameter (cm)  Circumference (cm)  Height (cm)  \
0   Architecture      885.1996           NaN                 NaN      48.6000   
1   Architecture        1.1995           NaN                 NaN      40.6401   
2   Architecture        1.1997           NaN                 NaN      34.3000   
3   Architecture        2.1995           NaN                 NaN      50.8000   
4   Architecture        2.1997           NaN                 NaN      38.4000   

   Length (cm)  Width (cm)  Depth (cm)  Weight (kg)  Duration (s)  
0          NaN    168.9000         NaN          NaN           NaN  
1          NaN     29.8451         NaN          NaN           NaN  
2          NaN     31.8000         NaN          NaN           NaN  
3          NaN     50.8000         NaN          NaN           NaN  
4          NaN     19.1000         NaN          NaN           NaN  

[5 rows x 21 columns]

In [19]:
artworks_python.isna().sum()

Artwork ID                 0
Title                     52
Artist ID               1460
Name                    1460
Date                    2312
Medium                 11919
Dimensions             11463
Acquisition Date        5463
Credit                  3070
Catalogue                  0
Department                 0
Classification             0
Object Number              0
Diameter (cm)         128863
Circumference (cm)    130252
Height (cm)            18369
Length (cm)           129526
Width (cm)             19259
Depth (cm)            118819
Weight (kg)           129964
Duration (s)          127178
dtype: int64

In [20]:
len(artworks_python)

130262

In [21]:
artists_python=pd.read_csv(r'C:\Users\Administrateur\Documents\artists.csv')
artists_python.head()

Artist ID             Name Nationality Gender  Birth Year  Death Year
0          1   Robert Arneson    American   Male      1930.0      1992.0
1          2   Doroteo Arnaiz     Spanish   Male      1936.0         NaN
2          3      Bill Arnold    American   Male      1941.0         NaN
3          4  Charles Arnoldi    American   Male      1946.0         NaN
4          5      Per Arnoldi      Danish   Male      1941.0         NaN

In [22]:
artists_python.isna().sum()

Artist ID          0
Name               0
Nationality     2488
Gender          3072
Birth Year      3854
Death Year     10512
dtype: int64

I created a SQLite database called Jolimoi.db on DB Browser for SQLite. 
I removed spaces in column names of artists table (Artist_ID, Birth_Year, Death_Year)

# Connecting SQL database to Python notebook

In [8]:
import sqlite3
conn = sqlite3.connect(r'C:\SQLite\Jolimoi.db')

In [9]:
tables = pd.read_sql("""SELECT *
                        FROM sqlite_master
                        WHERE type='table';""", conn)
tables

type      name  tbl_name  rootpage  \
0  table   artists   artists      8801   
1  table  artworks  artworks      8804   

                                                 sql  
0  CREATE TABLE "artists" (\n\t"Artist_ID"\tINTEG...  
1  CREATE TABLE "artworks" (\n\t"ArtworkID"\tINTE...

In [10]:
artists = pd.read_sql("""SELECT *
                        FROM artists LIMIT 5;""", conn)
artists

Artist_ID             Name Nationality Gender  Birth_Year  Death_Year
0          1   Robert Arneson    American   Male        1930      1992.0
1          2   Doroteo Arnaiz     Spanish   Male        1936         NaN
2          3      Bill Arnold    American   Male        1941         NaN
3          4  Charles Arnoldi    American   Male        1946         NaN
4          5      Per Arnoldi      Danish   Male        1941         NaN

In [11]:
artworks = pd.read_sql("""SELECT *
                        FROM artworks LIMIT 5;""", conn)
artworks

ArtworkID                                              Title  ArtistID  \
0          2  Ferdinandsbrücke Project, Vienna, Austria, Ele...      6210   
1          3  City of Music, National Superior Conservatory ...      7470   
2          4  Villa near Vienna Project, Outside Vienna, Aus...      7605   
3          5  The Manhattan Transcripts Project, New York, N...      7056   
4          6  Villa, project, outside Vienna, Austria, Exter...      7605   

                       Name  Date  \
0               Otto Wagner  1896   
1  Christian de Portzamparc  1987   
2                Emil Hoppe  1903   
3           Bernard Tschumi  1980   
4                Emil Hoppe  1903   

                                              Medium  \
0      Ink and cut-and-pasted painted pages on paper   
1                  Paint and colored pencil on print   
2  Graphite, pen, color pencil, ink, and gouache ...   
3  Photographic reproduction with colored synthet...   
4  Graphite, color pencil, ink, and gouache on tr...   

                           Dimensions AcquisitionDate  \
0  19 1/8 x 66 1/2" (48.6 x 168.9 cm)      1996-04-09   
1       16 x 11 3/4" (40.6 x 29.8 cm)      1995-01-17   
2   13 1/2 x 12 1/2" (34.3 x 31.8 cm)      1997-01-15   
3           20 x 20" (50.8 x 50.8 cm)      1995-01-17   
4    15 1/8 x 7 1/2" (38.4 x 19.1 cm)      1997-01-15   

                                              Credit Catalogue  ...  \
0  Fractional and promised gift of Jo Carole and ...         Y  ...   
1  Gift of the architect in honor of Lily Auchinc...         Y  ...   
2             Gift of Jo Carole and Ronald S. Lauder         Y  ...   
3  Purchase and partial gift of the architect in ...         Y  ...   
4             Gift of Jo Carole and Ronald S. Lauder         Y  ...   

  ObjectNumber Diameter(cm) Circumference(cm) Height(cm) Length(cm)  \
0     885.1996         None              None    48.6000       None   
1       1.1995         None              None    40.6401       None   
2       1.1997         None              None    34.3000       None   
3       2.1995         None              None    50.8000       None   
4       2.1997         None              None    38.4000       None   

   Width(cm) Depth(cm)  Weight(kg) Duration(s) Surface  
0   168.9000      None        None        None     0.0  
1    29.8451      None        None        None     0.0  
2    31.8000      None        None        None     0.0  
3    50.8000      None        None        None     0.0  
4    19.1000      None        None        None     0.0  

[5 rows x 22 columns]

# Question 1
Which artist in this data set lived the longest?

We are provided with columns Birth_Year and Death_Year in the table artists to calculate the number of years lived by artist.

We assume that the question takes into account only the dead artists (i.e. Death_Year column <> NaN).

Also, as I noticed some artists without birth year, I added a condition to be sure excluding them.

In [12]:
Question1 = pd.read_sql("""SELECT Name, Death_Year-Birth_Year as Years_lived
                        FROM artists
                        WHERE Birth_Year IS NOT NULL
                        AND Death_Year IS NOT NULL
                        ORDER BY Years_lived DESC 
                        LIMIT 5;""", conn)
Question1

Name  Years_lived
0  Union Paper Bag Machine Company, Philadelphia, PA          130
1                                      Leni Matthaei          108
2                                          B. Efimov          108
3                                 Manoel de Oliveira          107
4                                          Hans Erni          106

Note that the 1st line of the printed request doesn't seem to be an artist.

There are 2 artists with the same longest life of 108 years : Leni Matthaesi & B.Efimov

# Question 2
Who are the top 10 artists by the number of artworks?

In this question, the goal is to join the two tables in order to link Artists'names (from artists table) to their number of artworks (artworks table).
The primary key of both tables is ArtistID.

In [13]:
Question2 = pd.read_sql("""SELECT artt.Name, COUNT(artw.ArtworkID) as NberArtworks
                        FROM artists artt
                        JOIN artworks artw
                        ON artt.Artist_ID = artw.ArtistID
                        GROUP BY artt.Name
                        ORDER BY NberArtworks DESC
                        LIMIT 10
                        ;""", conn)
Question2

Name  NberArtworks
0              Eugène Atget          5050
1          Louise Bourgeois          3318
2  Ludwig Mies van der Rohe          2566
3      Unknown photographer          1575
4             Jean Dubuffet          1435
5           Lee Friedlander          1317
6             Pablo Picasso          1310
7              Marc Chagall          1162
8             Henri Matisse          1063
9            Pierre Bonnard           894

# Question 3
Which artist is created the most artwork by total surface area?

For this question, I need to calculate surface of each artwork. However, it's challenging as artworks can have various shapes (sculpture, paintings, objects...)

I need to take hypothesis to estimate sufaces :

- when there is a diameter, the artwork can be a circle (other dimensions columns null) OR a 3-dimension object (height column not null). To simplify my analysis, I will assume that those shapes are ovale.

- when there isn't diameter, I'm gonna assume (to simplify my analysis) that artworks are rectangular or square (most common shapes of artworks).


In [14]:
Question3 = pd.read_sql(""" SELECT Name, CASE 
                        WHEN "Diameter (cm)" NOT NULL 
                        AND "Height (cm)" IS NULL
                            THEN (3.14*(("Diameter (cm)"/2)*("Diameter (cm)"/2)))
                        WHEN "Diameter (cm)" NOT NULL 
                        AND "Height (cm)" NOT NULL
                            THEN (3.14*(("Diameter (cm)"/2)*("Height (cm)"/2)))
                        WHEN "Diameter (cm)" IS NULL 
                        AND "Height (cm)" IS NULL AND "Length (cm)" NOT NULL AND "Width (cm)" NOT NULL
                            THEN ("Length (cm)"*"Width (cm)")
                        WHEN "Diameter (cm)" IS NULL
                        AND "Length (cm)" IS NULL AND "Width (cm)" NOT NULL AND "Height (cm)" NOT NULL
                            THEN ("Width (cm)"*"Height (cm)")
                        WHEN "Diameter (cm)" IS NULL
                        AND "Width (cm)" IS NULL AND "Length (cm)" NOT NULL AND "Height (cm)" NOT NULL
                            THEN ("Length (cm)"*"Height (cm)")
                        ELSE 0 
                        END AS Surface
                        FROM artworks
                        ORDER BY Surface DESC
                        LIMIT 1;""", conn)
Question3

Name  Surface
0  Otto Wagner      0.0

In [55]:
# As my test gives me a 0 surface. However, when checking the data for this specific artist, I have data for Height and Width.
testotto = pd.read_sql(""" SELECT Name, "Diameter(cm)", "Height(cm)", "Length(cm)", "Width(cm)", "Depth(cm)"
from artworks
where Name="Otto Wagner";""", conn)
testotto

Name Diameter(cm)  Height(cm) Length(cm)  Width(cm)  Depth(cm)
0  Otto Wagner         None     48.6000       None   168.9000        NaN
1  Otto Wagner         None     78.5000       None    56.5000    51.5000
2  Otto Wagner         None     46.9901       None    40.6401    40.6401
3  Otto Wagner         None     72.4000       None   117.5000     7.6000

In [71]:
testotto2 = pd.read_sql(""" SELECT Name, SUM("Height(cm)"*"Width(cm)") as Surface
from artworks
where Name="Otto Wagner"
GROUP BY Name;""", conn)
testotto2

Name       Surface
0  Otto Wagner  23060.472363

In [44]:
test3 = pd.read_sql(""" PRAGMA table_xinfo(artworks);""", conn)
test3

cid               name     type  notnull dflt_value  pk  hidden
0     0          ArtworkID  INTEGER        0       None   0       0
1     1              Title     TEXT        0       None   0       0
2     2           ArtistID  INTEGER        0       None   0       0
3     3               Name     TEXT        0       None   0       0
4     4               Date     TEXT        0       None   0       0
5     5             Medium     TEXT        0       None   0       0
6     6         Dimensions     TEXT        0       None   0       0
7     7    AcquisitionDate     TEXT        0       None   0       0
8     8             Credit     TEXT        0       None   0       0
9     9          Catalogue     TEXT        0       None   0       0
10   10         Department     TEXT        0       None   0       0
11   11     Classification     TEXT        0       None   0       0
12   12       ObjectNumber     TEXT        0       None   0       0
13   13       Diameter(cm)     REAL        0       None   0       0
14   14  Circumference(cm)     REAL        0       None   0       0
15   15         Height(cm)     REAL        0       None   0       0
16   16         Length(cm)     REAL        0       None   0       0
17   17          Width(cm)     REAL        0       None   0       0
18   18          Depth(cm)     REAL        0       None   0       0
19   19         Weight(kg)     REAL        0       None   0       0
20   20        Duration(s)     REAL        0       None   0       0

# Question 4
Did any artists have artwork acquired during their lifetime?

I need to join the two tables and:

- for dead artists, verify if the year of AcquisitionDate of each artwork is between the Birth_Year and the Death_Year of the artist

- for artists alive (we assume that those who don't have a Death_Year are still alive), add them to the analysis

Note that from my exploratory analysis, I know that some artists don't have a Birth_Year and should be removed from this analysis. 

In [15]:
# DEAD artists who have artwork acquired during their lifetime
Question4a = pd.read_sql("""SELECT artt.Name
                        FROM artists artt
                        JOIN artworks artw
                        ON artt.Artist_ID = artw.ArtistID
                        WHERE strftime('%Y',artw.AcquisitionDate) <= artt.Death_Year
                        GROUP BY artt.Name
                        ;""", conn)
Question4a

Name
0       A. E. Gallatin
1       A. G. Fronzoni
2      A. M. Cassandre
3            Aarne Aho
4        Aaron Siskind
...                ...
2590    Édouard Boubat
2591    Édouard Pignon
2592  Édouard Vuillard
2593     Étienne Hajdu
2594  Öyvind Fahlström

[2595 rows x 1 columns]

In [62]:
# Number of dead artists who have artwork acquired during their lifetime
len(Question4a)

2595

In [ ]:
# Add artists alive (death year is null AND birth year is not null - as some artists don't have a birth year)
Question4b = pd.read_sql("""SELECT artt.Name
                        FROM artists artt
                        JOIN artworks artw
                        ON artt.Artist_ID = artw.ArtistID
                        WHERE (strftime('%Y',artw.AcquisitionDate) <= artt.Death_Year AND artt.Birth_Year IS NOT NULL)
                        OR (artt.Death_Year IS NULL AND artt.Birth_Year IS NOT NULL)
                        GROUP BY artt.Name
                        ;""", conn)
Question4b

In [ ]:
# Number of artists (both dead and alive) who have/had artwork acquired during their lifetime
len(Question4b)

# Question 5
Please review the quality of the data, and present any issues

#### Artists table
- A lot of missing values (Nationality 2488 rows, Gender 3072, Birth_Year 3854)

- For Birth_Year column we assume that artists without birth year should be eliminated from analysis including dates (Question 1 & 4)

#### Artworks table
- ArtistID rows with several artists (separated by a comma). For those ones, there is no join working with table artists (c.6500 rows)

- Cleaning to be done on the number of columns: some should be removed from the dataset as (i) there are too many missing values (% of total of column) or (ii) we don't need them in the questions asked. 

In [19]:
for i in artworks_python.columns:
    print(i,f'{100*artworks_python[i].isnull().sum()/artworks_python.shape[0]:.2f}%')

Artwork ID 0.00%
Title 0.04%
Artist ID 1.12%
Name 1.12%
Date 1.77%
Medium 9.15%
Dimensions 8.80%
Acquisition Date 4.19%
Credit 2.36%
Catalogue 0.00%
Department 0.00%
Classification 0.00%
Object Number 0.00%
Diameter (cm) 98.93%
Circumference (cm) 99.99%
Height (cm) 14.10%
Length (cm) 99.43%
Width (cm) 14.78%
Depth (cm) 91.22%
Weight (kg) 99.77%
Duration (s) 97.63%


# Question 6
Please group the artworks into as many clusters as you feel is appropriate, using attributes from both the artist and artworks tables, and assign each artwork to this new cluster.

#### Grouping by classification/department

In [129]:
#First, taking the artwork table, we can cluster artworks by department and/or classification. 
#The analysis will be exhaustive as there is no missing value in those two columns.
Question6a = pd.read_sql("""SELECT Classification, Department,
                        COUNT(ArtworkID) as Num_Artworks,
                        100*(COUNT(ArtworkID)/cast((SELECT COUNT(ArtworkID) FROM artworks) AS float)) as PerTotal
                        FROM artworks 
                        GROUP BY Classification
                        ORDER BY Num_Artworks Desc
                        limit 10
                    """, conn)
Question6a

Classification                  Department  Num_Artworks  \
0                      Print  Prints & Illustrated Books         30807   
1                 Photograph  Prints & Illustrated Books         29909   
2           Illustrated Book  Prints & Illustrated Books         26160   
3                    Drawing  Prints & Illustrated Books         11735   
4                     Design       Architecture & Design         11223   
5  Mies van der Rohe Archive       Architecture & Design          3331   
6               Architecture       Architecture & Design          2947   
7                      Video        Painting & Sculpture          2363   
8                       Film  Prints & Illustrated Books          2292   
9                   Painting        Painting & Sculpture          2270   

    PerTotal  
0  23.650028  
1  22.960649  
2  20.082603  
3   9.008767  
4   8.615713  
5   2.557154  
6   2.262364  
7   1.814036  
8   1.759531  
9   1.742642

#### Grouping by gender

In [130]:
# Grouping artworks by artists' gender (however, note that the analysis is not exhaustive as we have missing values in our dataset).
# For Male artists, there are 2 writings so we need to add a condition to group them in the same row.
Question6b = pd.read_sql("""SELECT CASE WHEN artt.Gender='male' THEN 'Male' ELSE artt.Gender END AS Gender,
                        COUNT(artw.ArtworkID) AS Num_Artworks,
                        (COUNT(artw.ArtworkID)/CAST((SELECT COUNT(ArtworkID) FROM artworks) AS float)*100) AS Per_Total
                        FROM artists artt
                        INNER JOIN artworks artw ON artt.Artist_ID=artw.ArtistID
                        GROUP BY 1
                        ORDER BY Num_Artworks Desc
                    """, conn)
Question6b

Gender  Num_Artworks  Per_Total
0    Male         99984  76.756076
1  Female         16193  12.431100
2    None          6196   4.756568

#### Grouping by nationality

In [132]:
# Grouping artworks by artists' nationality (being however that the analysis is not exhaustive as we have missing values )
Question6c = pd.read_sql("""SELECT artt.Nationality,
                        COUNT(artw.ArtworkID) AS Num_Artworks,
                        (COUNT(artw.ArtworkID)/CAST((SELECT COUNT(ArtworkID) FROM artworks) AS float)*100) AS Per_Total
                        FROM artists artt
                        INNER JOIN artworks artw ON artt.Artist_ID=artw.ArtistID
                        GROUP BY 1
                        ORDER BY Num_Artworks Desc
                        LIMIT 4
                    """, conn)
Question6c

Nationality  Num_Artworks  Per_Total
0    American         54545  41.873302
1      French         22303  17.121647
2      German          8987   6.899172
3     British          5500   4.222260

#### Grouping by gender & nationality

In [128]:
# Grouping artworks by gender and nationality
Question6d = pd.read_sql("""SELECT CASE WHEN artt.Gender='male' THEN 'Male' ELSE artt.Gender END AS Gender,
                        artt.Nationality,
                        COUNT(artw.ArtworkID) AS Num_Artworks,
                        (COUNT(artw.ArtworkID)/CAST((SELECT COUNT(ArtworkID) FROM artworks) AS float)*100) AS Per_Total
                        FROM artists artt
                        INNER JOIN artworks artw ON artt.Artist_ID=artw.ArtistID
                        GROUP BY 1, 2
                        ORDER BY Num_Artworks Desc
                        LIMIT 10
                    """, conn)
Question6d

Gender Nationality  Num_Artworks  Per_Total
0    Male    American         42473  32.605825
1    Male      French         21820  16.750856
2  Female    American         11139   8.551228
3    Male      German          7491   5.750718
4    Male     British          4921   3.777771
5    Male     Spanish          2902   2.227818
6    None        None          2664   2.045109
7    Male     Italian          2423   1.860097
8    Male    Japanese          1960   1.504660
9    Male       Swiss          1735   1.331931

#### Grouping by centuries of creation

I would like to group artworks by period. However, I've many cells with strings and not only a year with 4 digits. 
I'm more comfortable using Python regex to extract those dates

In [23]:
import re
artworks_python['Painting_Date']=artworks_python['Date'].str.extract(r"(\d{4})")
artworks_python.head()

Artwork ID                                              Title Artist ID  \
0           2  Ferdinandsbrücke Project, Vienna, Austria, Ele...      6210   
1           3  City of Music, National Superior Conservatory ...      7470   
2           4  Villa near Vienna Project, Outside Vienna, Aus...      7605   
3           5  The Manhattan Transcripts Project, New York, N...      7056   
4           6  Villa, project, outside Vienna, Austria, Exter...      7605   

                       Name  Date  \
0               Otto Wagner  1896   
1  Christian de Portzamparc  1987   
2                Emil Hoppe  1903   
3           Bernard Tschumi  1980   
4                Emil Hoppe  1903   

                                              Medium  \
0      Ink and cut-and-pasted painted pages on paper   
1                  Paint and colored pencil on print   
2  Graphite, pen, color pencil, ink, and gouache ...   
3  Photographic reproduction with colored synthet...   
4  Graphite, color pencil, ink, and gouache on tr...   

                           Dimensions Acquisition Date  \
0  19 1/8 x 66 1/2" (48.6 x 168.9 cm)       1996-04-09   
1       16 x 11 3/4" (40.6 x 29.8 cm)       1995-01-17   
2   13 1/2 x 12 1/2" (34.3 x 31.8 cm)       1997-01-15   
3           20 x 20" (50.8 x 50.8 cm)       1995-01-17   
4    15 1/8 x 7 1/2" (38.4 x 19.1 cm)       1997-01-15   

                                              Credit Catalogue  ...  \
0  Fractional and promised gift of Jo Carole and ...         Y  ...   
1  Gift of the architect in honor of Lily Auchinc...         Y  ...   
2             Gift of Jo Carole and Ronald S. Lauder         Y  ...   
3  Purchase and partial gift of the architect in ...         Y  ...   
4             Gift of Jo Carole and Ronald S. Lauder         Y  ...   

  Object Number Diameter (cm) Circumference (cm)  Height (cm)  Length (cm)  \
0      885.1996           NaN                NaN      48.6000          NaN   
1        1.1995           NaN                NaN      40.6401          NaN   
2        1.1997           NaN                NaN      34.3000          NaN   
3        2.1995           NaN                NaN      50.8000          NaN   
4        2.1997           NaN                NaN      38.4000          NaN   

   Width (cm)  Depth (cm)  Weight (kg)  Duration (s)  Painting_Date  
0    168.9000         NaN          NaN           NaN           1896  
1     29.8451         NaN          NaN           NaN           1987  
2     31.8000         NaN          NaN           NaN           1903  
3     50.8000         NaN          NaN           NaN           1980  
4     19.1000         NaN          NaN           NaN           1903  

[5 rows x 22 columns]

In [24]:
#Take an example with artist Kristen Hawthorne who has this date issue. Adding column painting date allows me to isolate creation date in a clean way.
artworks_python[artworks_python['Name']=="Kirsten Hawthorne"]

Artwork ID                                        Title Artist ID  \
8290        9370   Plate (folio 4) from FOUND MASKS 1975-1978      2551   
8291        9371                        Found Masks 1975-1978      2551   
8292        9372  Plate (folio 13) from FOUND MASKS 1975-1978      2551   
8293        9373  Plate (folio 14) from FOUND MASKS 1975-1978      2551   
8294        9374  Plate (folio 15) from FOUND MASKS 1975-1978      2551   
8295        9375  Plate (folio 16) from FOUND MASKS 1975-1978      2551   
8296        9376  Plate (folio 17) from FOUND MASKS 1975-1978      2551   
8297        9377  Plate (folio 18) from FOUND MASKS 1975-1978      2551   
8298        9378  Plate (folio 19) from FOUND MASKS 1975-1978      2551   
8299        9379  Plate (folio 20) from FOUND MASKS 1975-1978      2551   
8300        9380  Plate (folio 21) from FOUND MASKS 1975-1978      2551   
8301        9381  Plate (folio 22) from FOUND MASKS 1975-1978      2551   
8302        9382   Plate (folio 5) from FOUND MASKS 1975-1978      2551   
8303        9383  Plate (folio 23) from FOUND MASKS 1975-1978      2551   
8304        9384  Plate (folio 24) from FOUND MASKS 1975-1978      2551   
8305        9385  Plate (folio 25) from FOUND MASKS 1975-1978      2551   
8306        9386  Plate (folio 26) from FOUND MASKS 1975-1978      2551   
8307        9387  Plate (folio 27) from FOUND MASKS 1975-1978      2551   
8308        9388  Plate (folio 28) from FOUND MASKS 1975-1978      2551   
8309        9389  Plate (folio 29) from FOUND MASKS 1975-1978      2551   
8310        9390  Plate (folio 30) from FOUND MASKS 1975-1978      2551   
8311        9391  Plate (folio 31) from FOUND MASKS 1975-1978      2551   
8312        9392  Plate (folio 32) from FOUND MASKS 1975-1978      2551   
8313        9393   Plate (folio 6) from FOUND MASKS 1975-1978      2551   
8314        9394   Plate (folio 7) from FOUND MASKS 1975-1978      2551   
8315        9395   Plate (folio 8) from FOUND MASKS 1975-1978      2551   
8316        9396   Plate (folio 9) from FOUND MASKS 1975-1978      2551   
8317        9397  Plate (folio 10) from FOUND MASKS 1975-1978      2551   
8318        9398  Plate (folio 11) from FOUND MASKS 1975-1978      2551   
8319        9399  Plate (folio 12) from FOUND MASKS 1975-1978      2551   

                   Name                                       Date  \
8290  Kirsten Hawthorne                                  1975-1978   
8291  Kirsten Hawthorne  1978.  ("Objects/masks" found 1975-1978).   
8292  Kirsten Hawthorne                                  1975-1978   
8293  Kirsten Hawthorne                                  1975-1978   
8294  Kirsten Hawthorne                                  1975-1978   
8295  Kirsten Hawthorne                                  1975-1978   
8296  Kirsten Hawthorne                                  1975-1978   
8297  Kirsten Hawthorne                                  1975-1978   
8298  Kirsten Hawthorne                                  1975-1978   
8299  Kirsten Hawthorne                                  1975-1978   
8300  Kirsten Hawthorne                                  1975-1978   
8301  Kirsten Hawthorne                                  1975-1978   
8302  Kirsten Hawthorne                                  1975-1978   
8303  Kirsten Hawthorne                                  1975-1978   
8304  Kirsten Hawthorne                                  1975-1978   
8305  Kirsten Hawthorne                                  1975-1978   
8306  Kirsten Hawthorne                                  1975-1978   
8307  Kirsten Hawthorne                                  1975-1978   
8308  Kirsten Hawthorne                                  1975-1978   
8309  Kirsten Hawthorne                                  1975-1978   
8310  Kirsten Hawthorne                                  1975-1978   
8311  Kirsten Hawthorne                                  1975-1978   
8312  Kirsten Hawthorne                                  1975-1978   


In [25]:
artworks_python.groupby('Painting_Date').count()['Artwork ID']

Painting_Date
1501      1
1768      5
1786      1
1797     80
1799      1
       ... 
2013    430
2014    227
2015    211
2016     38
2017      2
Name: Artwork ID, Length: 201, dtype: int64

In [26]:
artw_dates = pd.pivot_table(artworks_python, index='Painting_Date', values='Artwork ID', aggfunc='count')
artw_dates.sort_values('Artwork ID', ascending=False)

Artwork ID
Painting_Date            
1966                 2432
1967                 2409
1968                 2340
1969                 2316
1965                 2312
...                   ...
1805                    1
1800                    1
1799                    1
1786                    1
1501                    1

[201 rows x 1 columns]

In [40]:
artw_dates2 = artw_dates.reset_index(level=0)
artw_dates2

Painting_Date  Artwork ID  Century
0            1501           1        1
1            1768           5        1
2            1786           1        1
3            1797          80        1
4            1799           1        1
..            ...         ...      ...
196          2013         430        5
197          2014         227        3
198          2015         211        3
199          2016          38        1
200          2017           2        1

[201 rows x 3 columns]

In [43]:
artw_dates2.dtypes

Painting_Date    object
Artwork ID        int64
Century           int64
dtype: object

In [46]:
artw_dates2['Painting_Date']=pd.to_numeric(artw_dates2['Painting_Date'])
artw_dates2

Painting_Date  Artwork ID  Century
0             1501           1        1
1             1768           5        1
2             1786           1        1
3             1797          80        1
4             1799           1        1
..             ...         ...      ...
196           2013         430        5
197           2014         227        3
198           2015         211        3
199           2016          38        1
200           2017           2        1

[201 rows x 3 columns]

In [47]:
artw_dates2.dtypes

Painting_Date    int64
Artwork ID       int64
Century          int64
dtype: object

In [48]:
century=[]
for year in artw_dates2['Painting_Date'] :
    if (year % 100 == 0):
        century.append(year // 100)
    else:
        century.append(year // 100+1)
        
artw_dates2['Century'] = century
print(artw_dates2)

     Painting_Date  Artwork ID  Century
0             1501           1       16
1             1768           5       18
2             1786           1       18
3             1797          80       18
4             1799           1       18
..             ...         ...      ...
196           2013         430       21
197           2014         227       21
198           2015         211       21
199           2016          38       21
200           2017           2       21

[201 rows x 3 columns]


In [54]:
artw_dates2.groupby('Century').sum()['Artwork ID']

Century
16         1
18        88
19      8432
20    106904
21     11386
Name: Artwork ID, dtype: int64